app/services/stat_builder.rb

```diff
@@ -0,0 +1,33 @@
+# typed: true
+# frozen_string_literal: true
+
+# Runs queries and builds the data structure for the summary table.
+class StatBuilder
+  extend T::Sig
+
+  sig { returns(T::Hash[Collection, Hash]) }
+  def self.build_stats
+    collections_by_id = Collection.all.index_by(&:id)
```

Comment:
Should we have performance concerns about `Collection.all` here?

Reply:
No, because the requirement is to show all the collections on the page.